core/src/mender-api.c

```diff
@@ -214,6 +214,38 @@ mender_api_perform_authentication(void) {
     return ret;
 }
 
+/**
+ * @see mender_http_perform()
+ */
+static mender_err_t
+authenticated_http_perform(char                *path,
```

Comment:
I'm guessing some of these arguments can be `const` pointers

Reply:
You'd think so. But they are passed to `mender_http_perform()` which doesn't take them as `const` (because they are passed to other functions missing `const`). So yes, they should be `const`, but that's a bit out of scope.